pandas/core/internals/blocks.py

```diff
@@ -3204,3 +3205,34 @@ def _putmask_preserve(nv, n):
         v = v.astype(dtype)
 
     return _putmask_preserve(v, n)
+
+
+def _convert_scalar_for_putitemlike(scalar, dtype: np.dtype):
```

Comment:
maybe in pandas.core.dtypes.cast ?

Reply:
Sure.  With convert_scalar out of libindex, we can move validate_numeric_casting to core.dtypes.cast as well (and after #31662 we'll be able to get rid of libindex.get_value_at, plus util.get_value_at and util.validate_indexer!)

Reply:
sgtm.